src/app/content/hooks/storeTextSize.ts

```diff
@@ -0,0 +1,37 @@
+import { ActionHookBody, AppServices, MiddlewareAPI } from '../../types';
+import { setTextSize } from '../actions';
+import { textResizerDefaultValue, textResizerStorageKey } from '../components/constants';
+import { textSize } from '../selectors';
+
+export const loadStoredTextSize = (services: MiddlewareAPI & AppServices) => async() => {
+  const { getState, dispatch } = services;
+  const state = getState();
+  let storedTextSize;
+  let value = textResizerDefaultValue;
+
+  if (state.content.textSize !== null) {
+    return;
+  }
+
+  if (typeof window !== 'undefined' && window.localStorage) {
+    storedTextSize = window.localStorage.getItem(textResizerStorageKey);
+  }
+
+  if (storedTextSize) {
+    value = parseInt(storedTextSize, 10);
+  }
+
+  dispatch(setTextSize(value));
```

Comment:
you wanna check that value is not NaN

Reply:
yeah this really did need better NaN and bounds checking, added that here and also in the event callback